hyperbolic/euclid/Line.py

```diff
@@ -64,6 +64,14 @@ def makePerpendicular(self, x, y, length=1):
     def makeParallel(self, x, y, length=1):
         perp = self.makePerpendicular(x, y)
         return perp.makePerpendicular(x, y, length)
+    def isPointOnSegment(self, x, y):
+        ''' Assumes that the given point is on the line '''
+        K1=(x-self.x1)*(self.x2-self.x1)+(y-self.y1)*(self.y2-self.y1)      #dot product of vector startPoint->(x,y) and startpoint->endPoint->(x,y)
+        K2=(x-self.x2)*(self.x1-self.x2)+(y-self.y2)*(self.y1-self.y2)      #dot product of vector endPoint->(x,y) and endpoint->startPoint->(x,y)
```

Comment:
When you have long comments, put them on the line before.  Also, did you mean `startpoint->endPoint` not `startpoint->endPoint->(x,y)`?
```suggestion
        # Dot product of vector startPoint->(x,y) and startpoint->endPoint
        K1=(x-self.x1)*(self.x2-self.x1)+(y-self.y1)*(self.y2-self.y1)
        # Dot product of vector endPoint->(x,y) and endpoint->startPoint
        K2=(x-self.x2)*(self.x1-self.x2)+(y-self.y2)*(self.y1-self.y2)
```